src/utils/ledger.js

```diff
@@ -122,23 +122,43 @@ if (IPC_RENDERER) {
      * @memberof Ledger
      * @inner
      */
-    sendTx(data, changeIndex, changeKeyIndex) {
+    sendTx(data, changeInfo, useOldProtocol) {
+      // XXX: if custom tokens not allowed, use old protocol for first change output
       // first assemble data to be sent
-      const arr = [];
-      if (changeIndex > -1) {
-        const changeBuffer = formatPathData(changeKeyIndex)
-        // encode the bit indicating existence of change and change path length on first byte
-        arr.push(hathorLib.transaction.intToBytes(0x80 | changeBuffer[0], 1))
-        // change output index on the second
-        arr.push(hathorLib.transaction.intToBytes(changeIndex, 1))
-        // Change key path of the address
-        arr.push(changeBuffer.slice(1));
+      let initialData;
+      if (useOldProtocol) {
+        // Old protocol
+        // Only allows 1 change output so the changeInfo[0] will be used if it exists
+        const arr = [];
+        if (changeInfo.length > 0) {
+          const change = changeInfo[0];
+          const changeBuffer = formatPathData(change.keyIndex)
+          // encode the bit indicating existence of change and change path length on first byte
+          arr.push(hathorLib.transaction.intToBytes(0x80 | changeBuffer[0], 1))
+          // change output index on the second
+          arr.push(hathorLib.transaction.intToBytes(change.outputIndex, 1))
+          // Change key path of the address
+          arr.push(changeBuffer.slice(1));
+        } else {
+          // no change output
+          arr.push(hathorLib.transaction.intToBytes(0, 1));
+        }
+        initialData = Buffer.concat(arr);
```

Comment:
We can move this outside this if/else like before, no? And move the `initialData` instantiation as well

Reply:
Done in [86d6ee0](https://github.com/HathorNetwork/hathor-wallet/pull/264/commits/86d6ee02563d136387db484d7848ea2da6a5a3a4)